chalice/deploy/swagger.py

```diff
@@ -176,10 +187,9 @@ def _add_view_args(self, apig_integ, view_args):
             for name in view_args
         ]
 
-    def _add_preflight_request(self, view, swagger_for_path):
-        # type: (RouteEntry, Dict[str, Any]) -> None
-        cors = view.cors
-        methods = view.methods + ['OPTIONS']
+    def _add_preflight_request(self, cors, methods, swagger_for_path):
+        # type: (CORSConfig, List[str], Dict[str, Any]) -> None
+        methods = methods + ['OPTIONS']
```

Comment:
I would be hesitant to do append as it may cause unintentional side effects if the methods list provided is used later on. I can do `+=` though.